pkg/lwip/contrib/sock/udp/lwip_sock_udp.c

```diff
@@ -120,23 +120,21 @@ ssize_t sock_udp_recv_buf(sock_udp_t *sock, void **data, void **ctx,
     if (remote != NULL) {
         /* convert remote */
         size_t addr_len;
-#if LWIP_IPV6
-        if (sock->base.conn->type & NETCONN_TYPE_IPV6) {
-            addr_len = sizeof(ipv6_addr_t);
-            remote->family = AF_INET6;
+        if (IS_ACTIVE(LWIP_IPV6)) {
+            if (!IS_ACTIVE(LWIP_IPV4) ||
+                (NETCONNTYPE_ISIPV6(sock->base.conn->type))) {
+                addr_len = sizeof(ipv6_addr_t);
+                remote->family = AF_INET6;
+            }
         }
```

Comment:
Shouldn't this be enough now?

```suggestion
        if ((NETCONNTYPE_ISIPV6(sock->base.conn->type))) {
            addr_len = sizeof(ipv6_addr_t);
            remote->family = AF_INET6;
        }
```

Reply:
Yes indeed!